extensions/positron-python/python_files/posit/positron/data_explorer.py

```diff
@@ -1925,8 +1934,22 @@ def _get_histogram_method(method: ColumnHistogramParamsMethod):
     }[method]
 
 
-def _get_histogram_numpy(data, num_bins, method="fd"):
-    import numpy as np
+def _get_histogram_numpy(data, num_bins, method="fd", *, to_numpy=False):
+    try:
+        import numpy as np
+    except ModuleNotFoundError as e:
+        # If NumPy is not installed, we cannot compute histograms
+        # intentionally printing since errors will not show up in the console
+        warnings.warn(
+            "Numpy not installed, histogram computation will not work. "
+            "Please install NumPy to enable this feature.",
+            category=DataExplorerWarning,
+            stacklevel=1,
+        )
```

Comment:
Could you add a test that uses [pytest.warns()](https://docs.pytest.org/en/stable/how-to/capture-warnings.html#warns) and maybe `monkeypatch.setitem(sys.modules, 'numpy', None)` to ensure we get a message here?